docs/api/synchronization_library/barrier.md

```diff
@@ -23,26 +23,35 @@ cuda::std::barrier<> bb;
 cuda::barrier<cuda::thread_scope_block> c;
 ```
 
-The class template `barrier` may also be declared without initialization; a
+The class template `barrier` may also be declared without initialization in the `cuda::` namespace; a
 friend function `init` may be used to initialize the object.
```

Comment:
```suggestion
Variable declarations whose type is an instantiation of the `barrier` class template may be initialized with the friend function `init`.
```

I think the "class template `barrier`" is already declared in the `::cuda` namespace and cannot be re-declared there. What can be declared without initialization are variables whose type is an instantiation of that template.

Reply:
```suggestion
Variables declared with a type that is an instantiation of the `cuda::barrier` class template may be declared without initialization; in such case, they must be initialized by calling a friend function `init` prior to any use.
```

How about the above?

Reply:
Looking at my suggested change now, I think I'll want to tweak further, but that's a thing for tomorrow.